commit 0b7201c1fe4f12dc7b17abbf19047864d1cca302
Author: Andrea Iacovitti <aiacovitti@libero.it>
Date:   Sat Jul 9 19:51:40 2011 +0200

    Fix regression about input text clearButton stopping to work on khtml forms
    BUG: 246513
    FIXED-IN: 4.7.0

diff --git a/kdeui/widgets/klineedit.cpp b/kdeui/widgets/klineedit.cpp
index 7382806..8f1c8a4 100644
--- a/kdeui/widgets/klineedit.cpp
+++ b/kdeui/widgets/klineedit.cpp
@@ -1116,7 +1116,7 @@ void KLineEdit::mousePressEvent( QMouseEvent* e )
     if  ( (e->button() == Qt::LeftButton ||
            e->button() == Qt::MidButton ) &&
           d->clearButton ) {
-        d->clickInClear = d->clearButton == childAt( e->pos() );
+        d->clickInClear = ( d->clearButton == childAt(e->pos()) || d->clearButton->underMouse() );
 
         if ( d->clickInClear ) {
             d->possibleTripleClick = false;
@@ -1145,7 +1145,7 @@ void KLineEdit::mousePressEvent( QMouseEvent* e )
 void KLineEdit::mouseReleaseEvent( QMouseEvent* e )
 {
     if ( d->clickInClear ) {
-        if ( d->clearButton == childAt( e->pos() ) ) {
+        if ( d->clearButton == childAt(e->pos()) || d->clearButton->underMouse() ) {
             QString newText;
             if ( e->button() == Qt::MidButton ) {
                 newText = QApplication::clipboard()->text( QClipboard::Selection );
